### PR TITLE
#62 Event 카테고리 Tag 이미지 셋팅

### LINKED ISSUE #62: Event 카테고리 Tag 이미지 셋팅

### PR DESCRIPTION
**Pull request by seunghee63**
- https://github.com/Nexters/talkbbokki-android/issues/62 
image
image

Closes #62

### CODE REVIEW

File: app/src/main/java/com/hammer/talkbbokki/presentation/main/MainScreen.kt [hunk 1]
```diff
@@ -159,15 +159,15 @@ fun MainHeader(
...
+    onClickLevel: (String, String) -> Unit,
     showDialog: () -> Unit
 ) {
     Box(modifier = Modifier.aspectRatio(1f)) {
         Card(
             modifier = Modifier.fillMaxSize(),
             shape = RoundedCornerShape(8.dp),
             onClick = {
-                if (level.isActive) onClickLevel(level.id) else showDialog()
+                if (level.isActive) onClickLevel(level.id, (level.title).replace("\n","**")) else showDialog()
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/main/MainScreen.kt:170]: 🚫 **[ktlint]** <max-line-length> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Exceeded max line length (100) (cannot be auto-corrected)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/main/MainScreen.kt:170]: 🚫 **[ktlint]** <comma-spacing> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing spacing after ","

File: app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt [hunk 2]
```diff
@@ -61,7 +61,7 @@ fun TalkbbokkiNavHost(
             navigateToDetail = { level, item ->
                 navController.navigate(
                     DetailDestination.route +
-                        "?level=$level&id=${item.id}&tag=${item.tag}&topic=${item.name}&shareLink=${item.shareLink}"
+                            "?level=$level&id=${item.id}&tag=${item.tag}&topic=${item.name}&shareLink=${item.shareLink}"
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt:64]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (28) (should be 24)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt:64]: 🚫 **[ktlint]** <max-line-length> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Exceeded max line length (100) (cannot be auto-corrected)

File: app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt [hunk 3]
```diff
@@ -77,7 +77,7 @@ fun TalkbbokkiNavHost(
             navigateToDetail = { item ->
                 navController.navigate(
                     DetailDestination.route +
-                        "?level=${item.category.uppercase(Locale.getDefault())}&id=${item.id}&tag=${item.tag}&topic=${item.name}&shareLink=${item.shareLink}"
+                            "?level=${item.category.uppercase(Locale.getDefault())}&id=${item.id}&tag=${item.tag}&topic=${item.name}&shareLink=${item.shareLink}"
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt:80]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (28) (should be 24)

Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/navigation/TalkbbokkiNavHost.kt:80]: 🚫 **[ktlint]** <max-line-length> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Exceeded max line length (100) (cannot be auto-corrected)

File: app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt [hunk 4]
```diff
@@ -45,12 +45,10 @@ fun TopicListRoute(
     onClickToDetail: (level: String, item: TopicItem) -> Unit,
     onClickToMain: () -> Unit,
     viewModel: TopicListViewModel = hiltViewModel(),
-    topicLevel: String
-) {
+    ) {
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListScreen.kt:49]: 🚫 **[ktlint]** <indent> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unexpected indentation (4) (should be 0)

File: app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt [hunk 5]
```diff
@@ -15,7 +16,9 @@ class TopicListViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
     private val topicUseCase: TopicUseCase
 ) : ViewModel() {
-    private val selectedLevel = savedStateHandle.get<String>("level") ?: "level1"
+    val selectedLevel = savedStateHandle.get<String>("level")?.toUpperCase() ?: "LEVEL1"
+    val selectedLevelTitle = savedStateHandle.get<String>("title") ?: ""
+
     val topicList: StateFlow<List<TopicItem>> = topicUseCase.invoke(selectedLevel)
```
Comment (Jiyoung9310) [inline app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: selectedLevel이 upperCase 로 바뀌면 api 호출이 제대로 안되지 않아? 이거 잘 동작 하는거양?

Comment (Jiyoung9310) [reply app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: 요거 ViewModel 말고 view 단에서 필요한 때에 upperCase로 바꾸게 처리 가능요? 선택한 level로 캐싱 되어있는지 아닌지 확인하는데 왠지 문제 생길 수도 있을 것 같아서

Comment (seunghee63) [reply app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: 아 캐싱처리 때문이면 ㅇㅋㅇㅋㅇㅋ !!! 수정할게유 ~

File: app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt [hunk 6]
```diff
@@ -1,5 +1,6 @@
 package com.hammer.talkbbokki.presentation.topics
 
+import androidx.compose.ui.text.toUpperCase
```
Comment (seunghee63) [inline app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt:3]: 🚫 **[ktlint]** <no-unused-imports> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import

File: app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt (hunk 5 above)
Comment (seunghee63) [reply app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: [83fb18e](https://github.com/Nexters/talkbbokki-android/pull/76/commits/83fb18ec75117cc15a6d3cc7fe5d68a02e958026)

File: app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt [hunk 7]
```diff
@@ -21,7 +21,7 @@ class TopicListViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
     private val topicUseCase: TopicUseCase
 ) : ViewModel() {
-    val selectedLevel = savedStateHandle.get<String>("level")?.toUpperCase() ?: "LEVEL1"
+    val selectedLevel = savedStateHandle.get<String>("level") ?: "LEVEL1"
```
Comment (Jiyoung9310) [inline app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: "LEVEL1" 요것두 "level1"로 바꿔주엉

Comment (seunghee63) [reply app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: 흐업.. 반영했서유 

[d04a3c7](https://github.com/Nexters/talkbbokki-android/pull/76/commits/d04a3c756104ea154f4ef06431810bb5491359eb)

Comment (Jiyoung9310) [reply app/src/main/java/com/hammer/talkbbokki/presentation/topics/TopicListViewModel.kt]: 귯~ 👍

**Review by Jiyoung9310 — APPROVED** [review]

**Pull request merged**